src/lib/operators/partitioning.cpp

```diff
@@ -0,0 +1,301 @@
+#include "partitioning.hpp"
+
+#include <map>
+
+#include "insert.hpp"
+#include "storage/partitioning/hash_partition_schema.hpp"
+#include "storage/partitioning/null_partition_schema.hpp"
+#include "storage/partitioning/range_partition_schema.hpp"
+#include "storage/partitioning/round_robin_partition_schema.hpp"
+#include "storage/storage_manager.hpp"
+#include "table_wrapper.hpp"
+
+namespace opossum {
+
+// We need these classes to perform the dynamic cast into a templated ValueColumn
+class AbstractTypedColumnProcessor {
+ public:
+  virtual void resize_vector(std::shared_ptr<BaseColumn> column, size_t new_size) = 0;
+  virtual void copy_data(std::shared_ptr<const BaseColumn> source, size_t source_start_index,
+                         std::shared_ptr<BaseColumn> target, size_t target_start_index,
+                         std::vector<size_t> rows_to_copy) = 0;
+};
+
+template <typename T>
+class TypedColumnProcessor : public AbstractTypedColumnProcessor {
```

Comment:
The following 80 lines of code are a 1:1 copy from the Insert operator -- maybe you can think of a way how to re-use the original code.